### PR TITLE
CI: update Fedora to 35

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Contains
- https://github.com/kubernetes-sigs/kind/pull/2525
for passing CI

### CODE REVIEW

**Comment by aojea** [comment]
does this mean that most fedora users has to enable this feature gate for podman rootless?
https://github.com/kubernetes-sigs/kind/issues/2524#issuecomment-965642432

**Comment by AkihiroSuda** [comment]
> does this mean that most fedora users has to enable this feature gate for podman rootless?

Yes, probably

File: pkg/cluster/internal/kubeadm/config.go [hunk 1]
```diff
@@ -448,6 +448,10 @@ func Config(data ConfigData) (config string, err error) {
 			return "", errors.Errorf("version %q is not compatible with rootless provider (hint: kind v0.11.x may work with this version)", ver)
 		}
 		data.FeatureGates["KubeletInUserNamespace"] = true
+
+		// For avoiding err="failed to get rootfs info: failed to get device for dir \"/var/lib/kubelet\": could not find device with major: 0, minor: 41 in cached partitions map"
+		// https://github.com/kubernetes-sigs/kind/issues/2524
+		data.FeatureGates["LocalStorageCapacityIsolation"] = false
```
Comment (aojea) [inline pkg/cluster/internal/kubeadm/config.go:454]: docker too?

Comment (AkihiroSuda) [reply pkg/cluster/internal/kubeadm/config.go:454]: What do you mean?

Comment (aojea) [reply pkg/cluster/internal/kubeadm/config.go:454]: sorry, I mean if this is a general problem for rootless or podman and rootless specific

Comment (AkihiroSuda) [reply pkg/cluster/internal/kubeadm/config.go:454]: General for all rootless containers

**Comment by aojea** [comment]
/lgtm
@BenTheElder for approval, I can't find a better solution than this, no ideal , but at least is constrained to rootless that we mark as experimental

**Comment by aojea** [comment]
/approve
This is contained under a flag for rootless

**Pull request merged**